### PR TITLE
Fit Camera to Feature

### LINKED ISSUE #2245: Replace fitBounds with fitFeature
"bounds" is an ill-defined concept in mapbox-gl-js due to pitch and rotation. A less ambiguous and more useful API would fit the map to an arbitrary GeoJSON feature. 

ref https://github.com/mapbox/mapbox-gl-js/issues/2112

### PR DESCRIPTION
**Pull request by asheemmamoowala**
Closes #2245.

### TODO
 - [ ] description
 - [ ] tests
 - [ ] documentation

### CODE REVIEW

File: src/ui/camera.js [hunk 1]
```diff
@@ -342,6 +342,65 @@ class Camera extends Evented {
         return this;
     }
 
+    /**
+     * Pans and zooms the map to contain its visible area within the geographical bounds of the feature geometry.
+     * This function will also reset the map's bearing to 0 if bearing is nonzero.
```
Comment (andrewharvey) [inline src/ui/camera.js]: does this mean fit to feature is pitch aware and will retain the existing pitch?

Comment (andrewharvey) [reply src/ui/camera.js]: > This function will also reset the map's bearing to 0 if bearing is nonzero.

What are the implications of this for backwards compatibility? In the future if we can support retaining the bearing as part of `fitFeature` would it just be a breaking change?

Comment (asheemmamoowala) [reply src/ui/camera.js]: It does not alter the pitch, but it also doesn't account for it and will not give the correct result. 
There is no resolution yet for the pitch aware fitting.

There is more work needed to maintain the bearing or change it to some other value as part of the fitting operation.

**Comment by navenedrob** [comment]
@ asheemmamoowala What is the status of this? I've run into a use case where this would be very valuable.

**Comment by mourner** [comment]
Closing as stale — feel free to reopen if there's still interest in shipping this.

**Pull request closed without merging**